packages/@aws-cdk/aws-synthetics/lib/canary.ts

```diff
@@ -309,6 +309,7 @@ export class Canary extends cdk.Resource {
    * Returns a default role for the canary
    */
   private createDefaultRole(prefix?: string): iam.IRole {
+    const { partition } = cdk.Stack.of(this);
```

Comment:
`stack.of` returns the encompassing stack, not the partition. This should be:

```
cdk.Stack.of(this).partition
``` 
Was this change successfully deployed?

Reply:
This is using [object destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#object_destructuring) to only select the partition from the returned stack object similar to [this code](https://github.com/aws/aws-cdk/blob/b96efd8aaa143845b9fe315a9ee1e8398c4d83c2/packages/%40aws-cdk/aws-route53-targets/lib/bucket-website-target.ts#L14). I've run the tests successfully but I've not deployed this exact CR to AWS Cloudformation. Could we merge without doing a deployment?

Reply:
How were you able to update the `integ.canary.expected.json` file without deploying it?

Reply:
I made the update based on similar changes I made to my Cloudformation project and deployed to AWS. I'm new to this Github repository so I'm not sure how to reference an unpublished version of AWS CDK from Github and deploy to my AWS account.

Reply:
This file needs to be updated by executing the integration test - **not manually**. The stack defined `integ.canary.ts` will be deployed and if successful the expectation file will be updated. 

From the @aws-cdk/aws-synthetics folder executing:

```
yarn integ integ.canary.js
```

Will deploy the stack and afterward update the file.

Reply:
Thanks for walking me through the process. I successfully deployed the changes to my AWS account and the integration test passed succesfully

```
gitpod /workspace/aws-cdk/packages/@aws-cdk/aws-synthetics $ yarn integ integ.canary.js      
  yarn run v1.22.5            
    $ cdk-integ integ.canary.js                  
      Synthesizing integ.canary.js.                        
        Selected stack: canary-one                              
          Deploying integ.canary.js...                                    
            Deployment succeeded, updating snapshot.
Cleaning up.
Done in 129.24s.
```

There were no additional changes after the snapshot was updated.